tests/pip_deps/WORKSPACE

```diff
@@ -0,0 +1,26 @@
+workspace(name = "pip_deps")
+
+local_repository(
+    name = "rules_python",
+    path = "../..",
+)
+
+load("@rules_python//python:repositories.bzl", "python_register_toolchains")
+
+# This toolchain is explicitly 3.10 while `rules_python` is 3.9 to act as
+# a regression test, ensuring 3.10 is functional
+python_register_toolchains(
+    name = "python310",
+    python_version = "3.10",
+)
+
+load("@python310//:defs.bzl", "interpreter")
+load("//:pip_deps.bzl", "pip_deps")
+
+PIP_PACKAGES = {"sphinx": "4.5.0"}
+
+pip_deps(
```

Comment:
this was the intent of the `pip_parse_vendored` example introduced in https://github.com/bazelbuild/rules_python/commit/12662b68d89976967a4b4653e79a9a88a6b29271

could you just update that example instead of introducing another e2e test? They are kinda slow so it would be better not to accumulate more.

Also this doc https://github.com/bazelbuild/rules_python/commit/12662b68d89976967a4b4653e79a9a88a6b29271#diff-b92f94a67df862cde9714ff224afa676e39fe02da4fd4c6993daf684fbbc7cbdR182-R186 should maybe be updated?

Reply:
> this was the intent of the `pip_parse_vendored` example introduced in [12662b6](https://github.com/bazelbuild/rules_python/commit/12662b68d89976967a4b4653e79a9a88a6b29271)
> 
The major issue with pip_parse_vendored and the approach used for golang is that the meta data is unknown to Bazel. With pip_deps, one can save all the metadata in a .bzl file so that the downstream can consume it, e.g., use it to resolve diamond dependency.

I did something similar with golang dependencies at https://github.com/BoleynSu-Org/monorepo/blob/main/configs/deps/go_deps.bzl.

> could you just update that example instead of introducing another e2e test? They are kinda slow so it would be better not to accumulate more.
>

Sure, I think I can merge it into another e2e test.
 
> Also this doc [12662b6#diff-b92f94a67df862cde9714ff224afa676e39fe02da4fd4c6993daf684fbbc7cbdR182-R186](https://github.com/bazelbuild/rules_python/commit/12662b68d89976967a4b4653e79a9a88a6b29271#diff-b92f94a67df862cde9714ff224afa676e39fe02da4fd4c6993daf684fbbc7cbdR182-R186) should maybe be updated?